File: backend/ws-service/src/main/java/com/example/wsservice/fcm/service/FcmService.java
package com.example.wsservice.fcm.service;

import com.example.wsservice.fcm.client.FcmFeignClient;
import com.example.wsservice.fcm.dto.FcmMessageDto;
import com.example.wsservice.fcm.dto.FcmUserMessageDto;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class FcmService {

    private final FcmFeignClient fcmFeignClient;

    public void sendFcmMessage(FcmMessageDto fcmMessageDto) {
        fcmFeignClient.sendMessage(fcmMessageDto);
    }

    public void sendUserMessage(FcmUserMessageDto fcmUserMessageDto) {
        fcmFeignClient.sendUserMessage(fcmUserMessageDto);
    }
}

File: backend/ws-service/src/main/kotlin/com/example/wsservice/fcm/service/FcmService.kt
package com.example.wsservice.fcm.service

import com.example.wsservice.fcm.client.FcmFeignClient
import com.example.wsservice.fcm.dto.FcmMessageDto
import com.example.wsservice.fcm.dto.FcmUserMessageDto
import org.springframework.stereotype.Service

@Service
class FcmService(private val fcmFeignClient: FcmFeignClient) {

    fun sendFcmMessage(fcmMessageDto: FcmMessageDto) {
        fcmFeignClient.sendMessage(fcmMessageDto)
    }

    fun sendUserMessage(fcmUserMessageDto: FcmUserMessageDto) {
        fcmFeignClient.sendUserMessage(fcmUserMessageDto)
    }
}
